src/Altinn.App.Core/Features/Signing/SigningService.cs

```diff
@@ -287,6 +259,135 @@ await organisationClient.GetOrganization(organisationSignee.OrganisationNumber)
+
+            SignDocument signDocument =
+                JsonSerializer.Deserialize<SignDocument>(signDocumentSerialized, _jsonSerializerOptions)
+                ?? throw new JsonException("Could not deserialize signature document.");
+
+            signDocuments.Add(signDocument);
+        }
+
+        return signDocuments;
+    }
+
+    /// <summary>
+    /// This method exists to ensure we have a SigneeContext for both signees that have been delegated access to sign and signees that have signed using access granted through the policy.xml file.
+    /// </summary>
+    private async Task SynchronizeSigneeContextsWithSignDocuments(
+        IInstanceDataMutator instanceMutator,
+        List<SigneeContext> signeeContexts,
+        List<SignDocument> signDocuments
+    )
+    {
+        foreach (SignDocument signDocument in signDocuments)
+        {
+            SigneeContext? matchingSigneeContext = signeeContexts.FirstOrDefault(x =>
+                x.PersonSignee?.SocialSecurityNumber == signDocument.SigneeInfo.PersonNumber
+                || x.OrganisationSignee?.OrganisationNumber == signDocument.SigneeInfo.OrganisationNumber
+            );
+
+            if (matchingSigneeContext is not null)
+            {
+                // If the signee has been delegated access to sign there will be a matching SigneeContext. Setting the sign document property on this context.
+                matchingSigneeContext.SignDocument = signDocument;
+            }
+            else
+            {
+                // If the signee has signed using access granted through the policy.xml file, there is no persisted signee context. We create a signee context on the fly.
+                Party party = await altinnPartyClient.LookupParty(
+                    new PartyLookup
+                    {
+                        Ssn = signDocument.SigneeInfo.PersonNumber,
+                        OrgNo = signDocument.SigneeInfo.OrganisationNumber,
+                    }
+                );
+
+                PersonSignee? personSignee = party.Person is not null
+                    ? new PersonSignee
+                    {
+                        SocialSecurityNumber = party.Person.SSN,
+                        DisplayName = party.Person.Name,
+                        FullName = party.Person.Name,
+                        OnBehalfOfOrganisation = party.Organization?.Name,
+                    }
+                    : null;
+
+                OrganisationSignee? organisationSignee = party.Organization is not null
+                    ? new OrganisationSignee
+                    {
+                        OrganisationNumber = party.Organization.OrgNumber,
+                        DisplayName = party.Organization.Name,
+                    }
+                    : null;
+
+                signeeContexts.Add(
+                    new SigneeContext
+                    {
+                        TaskId = instanceMutator.Instance.Process.CurrentTask.ElementId,
```

Comment:
Burde `SignDocument` også settes her?

Reply:
Jupp. Bug. Lager unittest for det caset også.